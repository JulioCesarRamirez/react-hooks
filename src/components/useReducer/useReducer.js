const initialState = [
  {
    id: 1,
    todo: 'Buy milk',
    done: false,
  },
];

const newTodo = {
  id: 2,
  todo: 'Buy bread',
  done: false,
};

const todoReducer = (state = initialState, action) => {
  if (action?.type === 'add') {
    return [...state, action.payload];
  }

  return state;
};

let todos = todoReducer();

const addTodoAction = {
  type: 'add',
  payload: newTodo,
};

todos = todoReducer(todos, addTodoAction);

console.log(todos);
