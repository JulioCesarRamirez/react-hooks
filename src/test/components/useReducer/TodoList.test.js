import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { demoTodos } from '../../fixtures/demoTodos';
import { TodoList } from '../../../components/useReducer/TodoList';

describe('Testing <TodoList/>', () => {
  const handleDelete = jest.fn();
  const handleToggle = jest.fn();
  const wrapper = shallow(
    <TodoList
      todos={demoTodos}
      handleDelete={handleDelete}
      handleDelete={handleToggle}
    />
  );
  it('should render correctly', () => {
    expect(wrapper).toThrowErrorMatchingSnapshot();
  });
  it('should have two <TodoListItem/>', () => {
    expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length);
    expect(wrapper.find('TodoListItem').at(0).prop('handleDelete')).toEqual(
      expect.any(Function)
    );
  });
});
