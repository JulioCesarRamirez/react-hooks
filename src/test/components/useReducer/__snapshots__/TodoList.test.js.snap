// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <TodoList/> should render correctly 1`] = `"received is not a function"`;
